### DISCUSSION TITLE
How to sign with USB token?

### DISCUSSION THREAD

**Discussion by duyngnth** [Keyfactor/signserver-ce#45] [Q&A]
I have installed SignServer on WSL2 and have adminweb running correctly.
Now I want to sign with my USB token, what can I do next? Thank you!
![signserver_adminweb](https://github.com/Keyfactor/signserver-ce/assets/102924671/639ac10c-cc95-43eb-9f1d-79c4a025b1c0)

**Comment by duyngnth** [comment]
@netmackan

**Comment by netmackan** [comment]
What USB token is that? Do you have a working PKCS#11 driver/shared library provided by the vendor?
If that is the case then it should be possible to configure SignServer to use that similarly to if you would have an actual HSM.

**Reply by duyngnth** [reply]
It's ePass2003 and I don't have the shared library
